src/pdfcommon.hpp

```diff
@@ -363,6 +363,7 @@ struct ShadingType2 {
     CapyPDF_DeviceColorspace colorspace;
     double x0, y0, x1, y1;
     CapyPDF_FunctionId function;
+    double d1, d2;
```

Comment:
Going through the spec this should be optional just like extend below. I changed `extend`s in trunk already, domain should be done in the same way.